packages/zquery/src/index.ts

```diff
@@ -114,170 +95,17 @@ export function createZQuery<
     | CreateZQueryUnaryProps<Name, State, DataType, FetchArgs>
     | CreateZQueryStreamingProps<Name, State, DataType, FetchArgs, ProcessedDataType>,
 ): ZQuery<Name, ProcessedDataType, FetchArgs> {
-  const { name, get, set, getUseStore } = props;
-
-  const setAbortController = (abortController: AbortController | undefined) => {
-    set(prevState => ({
-      ...prevState,
-      _zQueryInternal: {
-        ...prevState._zQueryInternal,
-        abortController,
-      },
-    }));
-  };
-
-  const incrementReferenceCounter = () => {
-    const newReferenceCount = get(getUseStore().getState())._zQueryInternal.referenceCount + 1;
-
-    set(prevState => ({
-      ...prevState,
-      _zQueryInternal: {
-        ...prevState._zQueryInternal,
-        referenceCount: newReferenceCount,
-      },
-    }));
-
-    return newReferenceCount;
-  };
-
-  const decrementReferenceCounter = () => {
-    const newReferenceCount = get(getUseStore().getState())._zQueryInternal.referenceCount - 1;
-
-    set(prevState => ({
-      ...prevState,
-      _zQueryInternal: {
-        ...prevState._zQueryInternal,
-        referenceCount: newReferenceCount,
-      },
-    }));
-
-    return newReferenceCount;
-  };
+  const { name, get, getUseStore } = props;
 
   return {
-    [`use${capitalize(name)}`]: (...args: FetchArgs) => {
-      const useStore = getUseStore();
-
-      useEffect(() => {
-        const fetch = get(useStore.getState())._zQueryInternal.fetch;
-
-        {
-          const newReferenceCount = incrementReferenceCounter();
-
-          if (newReferenceCount === 1) {
-            setAbortController(new AbortController());
-            void fetch(...args);
-          }
-        }
-
-        const onUnmount = () => {
-          const newReferenceCount = decrementReferenceCounter();
-
-          if (newReferenceCount === 0) {
-            get(useStore.getState())._zQueryInternal.abortController?.abort();
-            setAbortController(undefined);
-          }
-        };
-
-        return onUnmount;
-      }, [fetch]);
-
-      const returnValue = useStore(
-        useShallow(state => {
-          const zQuery = get(state);
-
-          return {
-            data: zQuery.data,
-            loading: zQuery.loading,
-            error: zQuery.error,
-          };
-        }),
-      );
-
-      return returnValue;
-    },
```

Comment:
This code (and the deleted code above it) is moved into `getUseHook()`, unchanged